src/Middleware/LoginSessionMiddleware.php

```diff
@@ -42,10 +43,12 @@ public function process(HTTPRequest $request, callable $delegate)
                 return $delegate($request);
             }
 
-            // Update LastAccessed date and IP address
-            $loginSession->LastAccessed = DBDatetime::now()->Rfc2822();
-            $loginSession->IPAddress = $request->getIP();
-            $loginSession->write();
+            // Update LastAccessed date and IP address if > that threshold
+            $date = DBDatetime::now()->Rfc2822();
+            $threshold = LoginSession::config()->last_accessed_threshold;
+            if (strtotime($date) > strtotime($loginSession->LastAccessed) + $threshold) {
+                $loginSession->updateLastAccessed($request);
+            }
```

Comment:
Have you done some profiling that shows this as providing an actual performance gain? I'd like to see some evidence that this is actually solving some problem or providing measurable and substantial performance gains before considering a change of this nature.

Reply:
@GuySartorelli a write request on the database, including any subrequests (eg: a form, displaying multiple grid fields that load data + make a search form request), has always a cost. if you have a mysql bin log enabled, it's going to be registered as a change on the database. it also means you can display pages in the cms with only read queries, which is significantly faster.
i think you can even probably see this for yourself quite easily by simply commenting the current write operation and see the difference it makes in the cms and all the subsequent ajax requests

Reply:
"a cost" could be a millisecond or a minute.... I don't have time to check this right now, and I don't know if I'll remember to get back to it. If you can provide some quantitative data that would go a long way to proving this change is worthwhile - otherwise it'll need to wait until I or someone else has time to check out what performance benefits this change might have.